lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -756,8 +773,44 @@ selectUTxO (Checkpoint wid sl _parent _bh) = fmap entityVal <$>
     selectList
         [ UtxoWalletId ==. wid
         , UtxoSlot ==. sl
+        , UtxoSlotSpent ==. Nothing
         ] []
 
+addSpentUtxos
+    :: [TxMeta]
+    -> [TxIn]
+    -> SqlPersistT IO ()
+addSpentUtxos metas txins = do
```

Comment:
This needs to be an update not an insert. Because the utxo that is spent should already be there.

For an incoming transaction, the utxo entries are already getting added as part of the checkpoint.
For outgoing transactions, we want to soft-delete the utxo that has been spent, which can be done by setting slot_spent.

With transactions, we change from Pending -> InChain by adding new rows, but for utxo we update the existing rows.

Reply:
ok. so let's say we have utxo0 at some slot0. if I understand well we want :

Reply:
now thinking that maximum one row per utxo is better option. and here, indeed, update will be used

Reply:
in the coming commit I will adopt this kind of behavior:
```
ok, so basically we have one row per utxo. it has this kind of lifetime:
(utxo0, slot0, Nothing) - slot0...slot2
(utxo0, slot0, (Just slot3)) for slot3 and later
```

Reply:
one row per utxo logic depicted above is now implemented. all tests adopted

Reply:
>  Then let's say slot incremented. now after putCheckpoint we want to have two rows:

No, if we create a new checkpoint with the same UTxO, we do not duplicate the UTxO since nothing has changed. So, the UTxO is pretty much immutable except for the `slotSpent` column which is populated when we spend the UTxO :+1: